inference-engine/src/auto_plugin/auto_exec_network.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <unordered_map>
+#include <map>
+#include <vector>
+#include <string>
+
+#include <cpp_interfaces/impl/ie_executable_network_thread_safe_default.hpp>
+#include <threading/ie_itask_executor.hpp>
+
+namespace AutoPlugin {
+
+using DeviceName = std::string;
+
+struct DeviceInformation {
+    DeviceName deviceName;
+    std::map<std::string, std::string> config;
+};
+
+class AutoExecutableNetwork : public InferenceEngine::ExecutableNetworkThreadSafeDefault {
+public:
+    using Ptr = std::shared_ptr<AutoExecutableNetwork>;
+
+    AutoExecutableNetwork(const InferenceEngine::ExecutableNetwork&                             network,
+                          const DeviceInformation&                                              deviceInfo,
+                          const bool                                                            needPerfCounters = false);
+
+    InferenceEngine::Parameter GetMetric(const std::string &name) const override;
+    InferenceEngine::IInferRequestInternal::Ptr CreateInferRequestImpl(InferenceEngine::InputsDataMap networkInputs,
+                                                                       InferenceEngine::OutputsDataMap networkOutputs) override;
+    InferenceEngine::RemoteContext::Ptr GetContext() const override;
+    ~AutoExecutableNetwork() override;
+
+    DeviceInformation                                            _deviceInfo;
+    InferenceEngine::ExecutableNetwork                           _network;
+    std::unordered_map<std::string, InferenceEngine::Parameter>  _config;
+    bool                                                         _needPerfCounters = false;
```

Comment:
what is about other methods like GetExecGraphInfo ?

Reply:
done